src/Field.php

```diff
@@ -539,7 +539,9 @@ public function isHidden(): bool
      */
     public function getCaption(): string
     {
-        return $this->caption ?? $this->ui['caption'] ?? $this->readableCaption($this->short_name);
+        $caption = $this->caption ?? $this->ui['caption'] ?? $this->readableCaption($this->short_name);
+
+        return $caption instanceof \Closure ? $caption($this) : $caption;
```

Comment:
strong usecase for caption as Closure?

Reply:
Same like this [comment](https://github.com/atk4/data/pull/683#discussion_r462558176)